### PR TITLE
CPM-603: Use UpsertProductCommand in Enrichment integration tests

### PR DESCRIPTION
**Pull request by cemorin**
Use UpsertProductCommand in Enrichment integration tests

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: tests/back/Pim/Enrichment/Integration/Storage/Sql/ProductModel/GetDescendantVariantProductIdentifiersIntegration.php [hunk 1]
```diff
@@ -25,6 +30,18 @@ protected function getConfiguration()
...
+        $this->get('pim_catalog.updater.attribute_option')->update($attributeOption, [
+            'code' => 'optionC',
+            'attribute' => 'a_simple_select',
+            'labels' => ['en_US' => 'C option'],
+        ]);
+        $constraints = $this->get('validator')->validate($attributeOption);
+        if (count($constraints) > 0) {
+            throw new \InvalidArgumentException((string)$constraints);
+        }
+        $this->get('pim_catalog.saver.attribute_option')->save($attributeOption);
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Integration/Storage/Sql/ProductModel/GetDescendantVariantProductIdentifiersIntegration.php:43]: Can you explain why you added that ?

File: tests/back/Pim/Enrichment/Integration/Storage/Sql/ProductModel/GetDescendantVariantProductIdentifiersIntegration.php [hunk 2]
```diff
@@ -25,6 +30,18 @@ protected function getConfiguration()
...
     {
+        $attributeOption = $this->get('pim_catalog.factory.attribute_option')->create();
+        $this->get('pim_catalog.updater.attribute_option')->update($attributeOption, [
+            'code' => 'optionC',
+            'attribute' => 'a_simple_select',
+            'labels' => ['en_US' => 'C option'],
+        ]);
+        $constraints = $this->get('validator')->validate($attributeOption);
+        if (count($constraints) > 0) {
+            throw new \InvalidArgumentException((string)$constraints);
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Integration/Storage/Sql/ProductModel/GetDescendantVariantProductIdentifiersIntegration.php:41]:
```suggestion
        $violations = $this->get('validator')->validate($attributeOption);
        if (count($violations) > 0) {
            throw new \InvalidArgumentException((string)$violations);
```

**Review by mmetayer — APPROVED** [review]

File: tests/back/Pim/Enrichment/Integration/Completeness/AbstractCompletenessTestCase.php [hunk 3]
```diff
@@ -90,41 +75,46 @@ protected function createAttribute(
...
 
-        return $product;
+    protected function getUserId(string $username): int
+    {
+        $query = <<<SQL
+            SELECT id FROM oro_user WHERE username = :username
+        SQL;
+        $stmt = $this->get('database_connection')->executeQuery($query, ['username' => $username]);
+        $id = $stmt->fetchOne();
+        Assert::assertNotNull($id);
```
[the remark this answers is not in the document]
Comment (cemorin) [reply tests/back/Pim/Enrichment/Integration/Completeness/AbstractCompletenessTestCase.php]: Done in all of them :ok_hand:

File: tests/back/Pim/Enrichment/Integration/Storage/Sql/ProductModel/GetDescendantVariantProductIdentifiersIntegration.php (hunk 1 above)
Comment (mmetayer) [reply tests/back/Pim/Enrichment/Integration/Storage/Sql/ProductModel/GetDescendantVariantProductIdentifiersIntegration.php]: crystal clear, thanks

**Review by Balamung — APPROVED** [review]

File: tests/back/Pim/Enrichment/Integration/Product/ConvertVariantToSimpleProductIntegration.php [hunk 4]
```diff
@@ -265,16 +243,7 @@ private function convertToSimpleProduct(ProductInterface $product): void
...
-    }
-
-    private function createProduct(string $identifier, array $data): ProductInterface
-    {
-        $product = $this->get('pim_catalog.builder.product')->createProduct($identifier);
-        $this->get('pim_catalog.updater.product')->update($product, $data);
-        $this->saveProduct($product);
-
-        return $product;
+        $this->upsertProduct($product->getIdentifier());
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Integration/Product/ConvertVariantToSimpleProductIntegration.php:246]: probably just 
```php
$this->upsertProduct($product->getIdentifier(), [new ConvertToSimpleProduct()]);

**Pull request merged**